### PR TITLE
implement offer crud + fix/refactoring serialization services

### PR DESCRIPTION
**Pull request by florianajir**

### CODE REVIEW

File: Manager/OfferManager.php [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    public $lastCount;
+    /**
+     * @var ApiProvider
+     */
+    private $apiProvider;
+    /**
+     * @var Serializer
+     */
+    private $serializer;
+    /**
```
Comment (unikoke94) [inline Manager/OfferManager.php]: missing new lines

File: Manager/OfferManager.php [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        int $size = self::DEFAULT_SIZE,
+        string $sort = self::DEFAULT_SORT,
+        string $status = self::DEFAULT_STATUS,
+        ?string $name = null,
+        ?string $text = null
+    ): array {
+        $this->lastCount = null;
+        $offers = [];
+        $query = array_filter(compact('from', 'size', 'sort', 'status', 'name', 'text'));
+        $apiResponse = $this->apiProvider->request('GET', '/api/offers', [RequestOptions::QUERY => $query]);
```
Comment (unikoke94) [inline Manager/OfferManager.php]: maybe use constants from `Symfony\Component\HttpFoundation\Request` for http methods

File: Manager/OfferManager.php [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+     * @param null|string $language
+     *
+     * @return Offer|null
+     */
+    public function get(string $slug, ?string $language = null): ?Offer
+    {
+        $offer = null;
+        $query = compact('language');
+        $apiResponse = $this->apiProvider->request(
+            'GET',
```
Comment (unikoke94) [inline Manager/OfferManager.php]: same as above (constants http method)

File: Manager/OfferManager.php [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    }
+
+    /**
+     * @param string $slug
+     *
+     * @return bool
+     */
+    public function delete(string $slug): bool
+    {
+        $apiResponse = $this->apiProvider->request('DELETE',
```
Comment (unikoke94) [inline Manager/OfferManager.php]: start on new line + same as above (http method constant)

File: Manager/OfferManager.php [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+    /**
+     * @param Offer $record
+     *
+     * @return Offer|null
+     */
+    public function create(Offer $record): ?Offer
+    {
+        $params = [RequestOptions::JSON => $this->serializer->normalize($record, null, ['write' => true])];
+        $apiResponse = $this->apiProvider->request('POST', '/api/offers', $params);
```
Comment (unikoke94) [inline Manager/OfferManager.php]: same as above (constants http method)

File: Manager/OfferManager.php [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    /**
+     * @param string $slug
+     * @param Offer  $record
+     *
+     * @return Offer
+     */
+    public function update(string $slug, Offer $record): Offer
+    {
+        $params = [RequestOptions::JSON => $this->serializer->normalize($record, null, ['write' => true])];
+        $apiResponse = $this->apiProvider->request('PUT', '/api/offers/'.$slug, $params);
```
Comment (unikoke94) [inline Manager/OfferManager.php]: same as above (constants http method)

File: Manager/PageManager.php [hunk 7]
```diff
@@ -144,8 +140,8 @@ public function get(string $slug, ?string $language = null): ?Page
                 RequestOptions::QUERY       => $query,
             ]);
         if ($apiResponse->getStatusCode() === Response::HTTP_OK) {
-            $data = json_decode($apiResponse->getBody(), true);
-            $page = $this->serializer->denormalize($data, Page::class);
+            /** @var Page $page */
+            $page = $this->serializer->denormalize(json_decode($apiResponse->getBody(), true), Page::class);
```
Comment (stvalett) [inline Manager/PageManager.php]: You can use your private method $this->denormalizeResponse() and directly return the result in this condition

**Pull request merged**